src/Npgsql/Replication/PgOutput/PgOutputReplicationOptions.cs

```diff
@@ -17,8 +17,9 @@ public class PgOutputReplicationOptions : IEquatable<PgOutputReplicationOptions>
     /// <param name="binary">Send values in binary representation</param>
     /// <param name="streaming">Enable streaming of in-progress transactions</param>
     /// <param name="messages">Write logical decoding messages into the replication stream</param>
-    public PgOutputReplicationOptions(string publicationName, ulong protocolVersion, bool? binary = null, bool? streaming = null, bool? messages = null)
-        : this(new List<string> { publicationName ?? throw new ArgumentNullException(nameof(publicationName)) }, protocolVersion, binary, streaming, messages)
+    /// <param name="twoPhase">Enable streaming of prepared transactions</param>
+    public PgOutputReplicationOptions(string publicationName, ulong protocolVersion, bool? binary = null, bool? streaming = null, bool? messages = null, bool? twoPhase = null)
```

Comment:
Maybe we're at the point where we should introduce a [Flags] enum? :sweat_smile: Four bools seems like a lot (we could obsolete the previous one and make it call into the new one)

Reply:
> Four bools seems like a lot

I agree except that they aren't `bool` but `bool?` which is quite important here.
We are not talking about flags (that can be present or missing) but boolean options (that can be `true`, `false` or missing) here.
We use the nullabillity feature here to transfer the information to not set this variable at all (neither to `true` nor to `false`) and just keep the default (whatever it is, and however it might change by future decisions or configurations).

This doesn't translate well to a `[Flags]` enum. 

We may discuss whether this behavior makes sense or is worthwhile but we need to be aware of the fact that changing this also changes the behavior.

Personally for now I'd rather keep it the way it is.
The bools have default values so you don't have to worry about them if you don't care and personally I find setting named arguments even a bit more convenient than OR-ing flags.

Reply:
Makes sense, I think I missed the null aspect.